src/components/Form/Form.tsx

```diff
@@ -0,0 +1,69 @@
+import { useForm, FieldValues } from "react-hook-form";
+
+interface FormProps {
+  className?: string;
+  children?: React.ReactNode;
+  onSubmit: (data: FieldValues) => void;
+}
+
+export const Form = ({ className, children, ...rest }: FormProps) => {
+  return (
+    <form className={`flex flex-col gap-[7.47px] ${className}`} {...rest}>
+      {children}
+    </form>
+  );
+};
+
+interface TextInputProps {
+  placeholder?: string;
+  type: "email" | "tel";
+  register: ReturnType<typeof useForm>["register"];
+  errors: ReturnType<typeof useForm>["formState"]["errors"];
+  handleSubmit: ReturnType<typeof useForm>["handleSubmit"];
+  onSubmit: (data: FieldValues) => void;
+}
+
+Form.TextInput = ({
+  placeholder,
+  type,
+  register,
+  errors,
+  handleSubmit,
+  onSubmit,
+}: TextInputProps) => {
+  const handleEnter = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      handleSubmit(onSubmit)();
+    }
+  };
+
+  const validationRules = {
+    required: type === "tel" ? "Phone number is required" : "Email is required",
+    pattern: {
+      value:
+        type === "tel"
+          ? /^[+]?[0-9\s\-().]{10,15}$/
+          : /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
+      message:
+        type === "tel" ? "Invalid phone number format" : "Invalid email format",
+    },
+  };
+  return (
+    <div className="flex flex-col ">
+      <input
+        className={`w-[411.03px] h-[47.04px] p-[12.7px] bg-offwhite rounded-[12px] text-form-text bg-offWhite  ${
```

Comment:
One small change might be to use the red in tailwind, I would suggest using "gingerFlower" since it looked like brightest red which would be good for an error. 
I also noticed that there is a red border when there is an error, but it is covered by the default blue border. I tried to figure out how to override the blue border from the default but didn't figure it out. 
Lastly, could the red error border be thicker like "border-2"?
<img width="1080" alt="Screenshot 2024-09-15 at 8 47 41 AM" src="https://github.com/user-attachments/assets/874945bc-66d6-475b-a12b-1abe9b1071d6">
<img width="1080" alt="Screenshot 2024-09-15 at 8 47 08 AM" src="https://github.com/user-attachments/assets/1b69f58f-354f-4c4a-bb7f-bc0949b2f0db">

Reply:
Optional suggestion... Another thought is maybe to override the default blue border with a grey color. There is a grey in the tailwind called "lightGrey" that would work. The figma doesn't specify how to style an active input but in my opinion the blue clashes with the design. A grey border would be a nice way to highlight the input when active.